WcaOnRails/app/models/registration.rb

```diff
@@ -231,6 +231,16 @@ def self.wcif_json_schema
     end
   end
 
+  validate :cannot_register_for_unqualified_events
+  private def cannot_register_for_unqualified_events
+    if competition && competition.allow_registration_without_qualification
+      return
+    end
+    if registration_competition_events.reject(&:marked_for_destruction?).select { |event| event.competition_event.nil? || !event.competition_event.can_register?(user) }.any?
```

Comment:
In contrast to #7050 (which has a very strong motivation because the WCIF PATCH endpoint is even exposed to the OAuth public!!), deleting events for which competitors have already registered can sometimes happen. So your code should be able to deal with it anyways.

But interesting how `registration_competition_events` does not seem to be updated when overall events are deleted... Maybe this will actually be implicitly fixed by #7050, who knows :thinking: 

In terms of this change, what about `!event.competition_event&.can:register?(user)`? The `&` is a nil-safe operator in Ruby, and if the event doesn't exist it breaks out of the call chain and returns nil which then gets negated to `true` so it ultimately gets rejected.

Reply:
Ah thank you!